book/src/developers/designs/messaging.md

```diff
@@ -0,0 +1,99 @@
+# Messaging
+
+A number of features require us to be able to communicate with users. This includes things like user
+account signups, credential reset requests, and notification of security events.
+
+Each of these has unique requirements, so simply sending an email is not sufficent.
+
+## Use Cases
+
+This is not exhaustive, but covers some important and interesting use cases that affect our design
+decisions.
+
+### User Account Signups
+
+During an account signup, because the user is *new*, we don't have access to their mail attribute
+via an account object or similar. Because of this we need to ensure that when we send mail, it doesn't
+collide with existing accounts who own an email address, but also that we store the email in the requested
```

Comment:
"but also that we store the email in the requested message send."

I've tried to read this a few times and can't make sense of the words in this order. Can you rephrase it? 😄